parsers/DK.py

```diff
@@ -122,11 +148,47 @@ def fetch_exchange(
                     tzinfo=timezone.utc
                 ),
                 netFlow=flow(sorted_keys, datapoint),
-                source="energidataservice.dk",
+                source=SOURCE,
             )
         return all_exchange_data.to_list()
 
 
+@refetch_frequency(timedelta(days=1))
+def fetch_wind_solar_forecasts(
+    zone_key: ZoneKey,
+    session: Session | None = None,
+    target_datetime: datetime | None = None,
+    logger: Logger = getLogger(__name__),
+) -> list:
+    data = fetch_data(zone_key, session, target_datetime, logger, True)
+
+    # Group data by datetime Minutes5UTC
+    data["records"].sort(key=lambda x: x["Minutes5UTC"])
+    grouped_data = groupby(data["records"], key=lambda x: x["Minutes5UTC"])
```

Comment:
```suggestion
    grouped_data = groupby(data["records"], key=itemgetter("Minutes5UTC"))
```
There should be no need to pre-sort the data as it's sorted by the internal functions.
And using a itemgetter instead of a lambda for simple property selection slightly more efficient as it uses C code internally and don't need to create a function on each iteration (the lambda).

Reply:
Updated